### ISSUE TITLE
Library autodetection doesn't work when one library uses another library

### ISSUE THREAD

**Issue by matthijskooijman** [sudar/Arduino-Makefile#93]
I'm using the ShiftOutX library, which uses the SPI library. Arduino-Makefile fails to detect this case, resulting in a SPI.h not found error message.

**Comment by pglu** [comment]
Hello, is this issue still unsolved?

**Comment by xxxajk** [comment]
I'm happy to see someone else has forked the project!
That said...
add this to the top of your .ino, and it will work IDE or not: `#include <Arduino.h>`

After you do that, it *will* detect everything, At least it does in my trunk, which this fork is based on...

**Comment by pglu** [comment]
Hi, thanks for your quick reply, I am also happy to be able to use your project. However I'm still struggling with some libraries, but will check out including Arduino.h. The reason may be, that on my Raspi IDE 1.0.5 is instalIed don't know how to update it.

**Comment by sej7278** [comment]
No need to include Arduino.h, read the source, that's already done.

And this isn't a fork of anything from xxxajk, don't feed the troll

**Comment by xxxajk** [comment]
While this isn't a fork of my effort, both have a common root, which IIRC is pre-github days.

**Comment by sej7278** [comment]
Yup, first version of this we have history of this was [2010](https://github.com/sudar/Arduino-Makefile/blob/master/HISTORY.md#2010-05-21-version-03)
